src/main/java/com/linecorp/armeria/client/routing/Node.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import static java.util.Objects.requireNonNull;
+
+public class Node {
+
+    private String name;
+
+    private int port;
+
+    private int weight = DEFAULT_WEIGHT;
```

Comment:
Could you add the type hierarchy as I proposed? You could also propose something better.